pkg/addr/addr.go

```diff
@@ -29,16 +30,16 @@ func PublicAddressToString(addr *l5dNetPb.TcpAddress) string {
 
 // PublicIPToString formats a Viz API IPAddress as a string.
 func PublicIPToString(ip *l5dNetPb.IPAddress) string {
-	var b []byte
+	var netIP net.IP
 	if ip.GetIpv6() != nil {
-		b = make([]byte, 16)
-		binary.BigEndian.PutUint64(b[:8], ip.GetIpv6().GetFirst())
-		binary.BigEndian.PutUint64(b[8:], ip.GetIpv6().GetLast())
+		netIP = decodeIPv6ToNetIP(ip.GetIpv6())
 	} else if ip.GetIpv4() != 0 {
-		b = make([]byte, 4)
-		binary.BigEndian.PutUint32(b, ip.GetIpv4())
+		netIP = decodeIPv4ToNetIP(ip.GetIpv4())
 	}
-	return net.IP(b).String()
```

Comment:
hi @zhlsunshine, thanks for this fix!  Could you help me understand what caused this bug in the original code?  Originally, we would simply write the high and low halves of the ipv6 address (as uint64s) into a 16 byte array.  What does a : prefix address look like in that representation and why does `net.IP(b).String()` handle it incorrectly?

In the fixed code, it seems like there are quite a few steps:
* We write the high and low halves of the ipv6 address into a 16 byte array (b)
* We create a bigInt with those bytes
* We take the bytes back out of the bigInt (ipv6Bytes)
* We create a new byte buffer (buf) and zero pad it up the ipv6 length, then write the bytes into it
* We copy the buf into a new net.IP

Are all of these steps necessary?  Some seem a bit redundant.  In particular, our byte array (b) is already the full 16 bytes to begin with.  Do the bytes in (b) simply need to be shifted in certain cases?

Reply:
Hi @adleong, thank you for your comment. Yes, you are correct, it's redundant here. However, I am thinking that we may represent an IPv6 address into only one `big.Int` field instead separating it into 2 halves uint64 which may caused cumbersome in later.
BTW, respectfully, I want to submit a proposal and do some API change, but I do not exactly know the process. Moreover, it should not be a good title for this PR. Of course, if you do not want to modify the ipv6 API, I also can change the function `decodeIPv6ToNetIP` as below and remove the function `IntToIPv6` :

```
func decodeIPv6ToNetIP(ip *l5dNetPb.IPv6) net.IP {
        b := make([]byte, 16)
        binary.BigEndian.PutUint64(b[:8], ip.GetFirst())
        binary.BigEndian.PutUint64(b[8:], ip.GetLast())
        return net.IP(b)
}
```
Any comment?